packages/synapse-react-client/src/components/CreateTableViewWizard/CreateTableViewWizard.tsx

```diff
@@ -0,0 +1,372 @@
+    return false
+  }, [
+    entityViewScopeIds.length,
+    submissionViewScopeIds.length,
+    step,
+    viewTypeMask,
+  ])
+
+  const stepContent = useMemo(() => {
+    switch (step) {
+      case 'CHOOSE_TABLE_TYPE':
+        return (
+          <TableTypeSelection
+            onTableSelected={() => {
+              onTableTypeSelected(EntityType.TABLE)
+            }}
+            onViewSelected={onViewSelected}
+          />
+        )
+      case 'CHOOSE_VIEW_TYPE':
+        return <ViewTypeSelection onTypeSelected={onTableTypeSelected} />
+      case 'ENTITY_VIEW_SCOPE':
+        return (
+          <>
+            <EntityViewScopeEditor
+              scopeIds={entityViewScopeIds}
+              onChange={setEntityViewScopeIds}
+              isProjectView={isProjectView}
+            />
+            {!isProjectView && (
+              <EntityViewMaskEditor
+                value={viewTypeMask}
+                onChange={setViewTypeMask}
+              />
+            )}
+          </>
+        )
+      case 'SUBMISSION_VIEW_SCOPE':
+        return (
+          <SubmissionViewScopeEditor
+            evaluationIds={submissionViewScopeIds}
+            onChange={setSubmissionViewScopeIds}
+          />
+        )
+      case 'TABLE_SQL':
+        return (
+          <>
+            <TableNameForm
+              name={name}
+              setName={setName}
+              description={description}
+              setDescription={setDescription}
+            />
+            {/* TODO: Move SqlDefinedTableEditor to its own step, see https://sagebionetworks.jira.com/browse/PLFM-8209 */}
+            <SqlDefinedTableEditor
+              value={sql}
+              onChange={e => setSql(e.target.value)}
+              entityType={entityType}
+            />
+          </>
+        )
+      case 'TABLE_COLUMNS':
+        return (
+          <TableColumnSchemaForm
+            initialData={columnModels}
+            entityType={entityType!}
+            ref={columnSchemaFormRef}
+            onSubmit={onColumnSchemaSubmit}
+            viewScope={viewScope}
+          />
+        )
+      case 'TABLE_NAME':
+        return (
+          <TableNameForm
+            name={name}
+            setName={setName}
+            description={description}
+            setDescription={setDescription}
+          />
+        )
+    }
+  }, [
+    columnModels,
+    description,
+    entityType,
+    entityViewScopeIds,
+    isProjectView,
+    name,
+    onColumnSchemaSubmit,
+    onTableTypeSelected,
+    onViewSelected,
+    sql,
+    step,
+    submissionViewScopeIds,
+    viewTypeMask,
+  ])
+
+  return (
+    <DialogBase
+      open={open}
+      onCancel={onCancel}
+      maxWidth={'md'}
+      title={getModalTitle(step, entityType)}
+      content={
+        <div>
+          {stepContent}
+          {createEntityError && (
+            <Alert sx={{ my: 2 }} severity="error">
+              {createEntityError.message}
+            </Alert>
+          )}
```

Comment:
Currently, an error generated while writing defining SQL for a view will be shown on previous screens as well. Per @nickgros, instead plan to only shown errors on the screen where the call is made